keel-core/src/main/kotlin/com/netflix/spinnaker/keel/validators/DeliveryConfigValidator.kt

```diff
@@ -83,5 +86,20 @@ class DeliveryConfigValidator {
         }
       }
     }
+
+    /**
+     * check: all referenced artifacts exist
+     */
+    config.environments.forEach { environment ->
+      environment.resources.forEach { resource ->
+        (resource.spec as? ArtifactReferenceProvider)
+          ?.artifactReference
+          ?.also {
+            if (!refs.contains(it)) {
+              throw NonexistentArtifactReferenceException(it)
```

Comment:
Consider fixing that name as you're touching this:
```suggestion
              throw NonExistentArtifactReferenceException(it)
```

Reply:
I ended up renaming this to `InvalidArtifactReferenceException` in a653ee5afe06df3356d24b31a683950182761c1f